pkg/batcher/batcher.go

```diff
@@ -114,16 +114,32 @@ func OneBucketHasher[T input](ctx context.Context, input *T) uint64 {
 }
 
 func (b *Batcher[T, U]) run() {
```

Comment:
This is really carefully crafted code. Why are we fundamentally changing the batcher as part of this large PR?

Reply:
I'm putting the PR back in draft. This branch has become my staging environment for other changes. There is a separate PR for getting parallel batching. I agree that I wouldn't want to put all these changes together in a single PR